addons/im_livechat/static/src/core/channel_member_list_patch.js

```diff
@@ -3,8 +3,13 @@
 import { ChannelMemberList } from "@mail/discuss/core/common/channel_member_list";
 import { patch } from "@web/core/utils/patch";
 
+ChannelMemberList.props.push("openChannelInvitePanel?");
 patch(ChannelMemberList.prototype, "im_livechat", {
     canOpenChatWith(member) {
         return this._super(member) && !member.persona.is_public;
     },
+
+    onClickAddUser() {
+        this.props.openChannelInvitePanel?.(this.env.inChatWindow);
```

Comment:
Isn't `openChannelInvitePanel` necessarily defined when the button is present/clickable?

Does it actually need to be passed as props, rather than just being taken here from the registry?

Reply:
It's not as simple as taking it from the registry:
- The registry contains *untransformed* actions.
- The owner of the *transformed* action is the container itself since it's returned by a hook (`ChatWindow`/`Discuss`).
- The action components are instanciated via `t-component` so it would be cumbersome to pass this props directly from the parent as we would need to add a condition on whether the current action is the channel member one.

As for the optional chaining part you are right. Updated :+1: